tests/arp/test_neighbor_mac_noptf.py

```diff
@@ -0,0 +1,247 @@
+import logging
+import pytest
+import time
+
+from common.helpers.assertions import pytest_assert
+
+logger = logging.getLogger(__name__)
+
+class TestNeighborMacNoPtf:
+    """
+        Test handling of neighbor MAC in SONiC switch
+    """
+    TEST_MAC = {
+        4: ["08:bc:27:af:cc:45", "08:bc:27:af:cc:47"],
+        6: ["08:bc:27:af:cc:65", "08:bc:27:af:cc:67"],
+    }
+
+    TEST_INTF = {
+        4: {"intfIp": "29.0.0.1/24", "NeighborIp": "29.0.0.2"},
+        6: {"intfIp": "fe00::1/64", "NeighborIp": "fe00::2"},
+    }
+
+    @pytest.fixture(scope="module", autouse=True)
+    def restoreDutConfig(self, duthost):
+        """
+            Restores DUT configuration after test completes
+
+            Args:
+                duthost (AnsibleHost): Device Under Test (DUT)
+
+            Returns:
+                None
+        """
+        yield
+ 
+        logger.info("Reload Config DB")
+        duthost.shell(argv=["config", "reload", "-y"])
+ 
+        time.sleep(120)
+
+    @pytest.fixture(params=[4, 6])
+    def ipVersion(self, request):
+        """
+            Parameterized fixture for IP versions. This Fixture will run the test twice for both
+            IPv4 and IPv6 
+
+            Args:
+                request: pytest request object
+
+            Retruns:
+                ipVersion (int): IP version to be used for testing
+        """
+        yield request.param
+
+    @pytest.fixture(scope="module")
+    def routedInterface(self, duthost):
+        """
+            Find routed interface to test neighbor MAC functionality with
+
+            Args:
+                duthost (AnsibleHost): Device Under Test (DUT)
+
+            Retruns:
+                routedInterface (str): Routed interface used for testing
+        """
+        testRoutedInterface = None
+
+        intfStatus = duthost.show_interface(command="status")["ansible_facts"]["int_status"]
+        for intf, status in intfStatus.items():
+            if "routed" in status["vlan"] and "up" in status["oper_state"]:
+                testRoutedInterface = intf
+        pytest_assert(testRoutedInterface, "Failed to find a routed interface in '%s'" % intfStatus)
+
+        yield testRoutedInterface
+
+    @pytest.fixture
+    def verifyOrchagentPresence(self, duthost):
+        """
+            Verify orchagent is running before and after the test is finished
+
+            Args:
+                duthost (AnsibleHost): Device Under Test (DUT)
+
+            Returns:
+                None
+        """
+        def getOrchagentPid(duthost):
```

Comment:
This is not a getter? it didn't return anything.

Reply:
good catch. Will rename. I had in mind to return the pid and have the caller asserts, then relocated the assert!